File: backend/model/surat.go
package model

type Surat struct {
	Id              uint   `json:"id" gorm:"primaryKey"`
	NomorSurat      string `json:"nomor_surat" gorm:"unique;not null"`
	Status          string `json:"status" gorm:"not null"`
	TanggalBuat     string `json:"tanggal_buat" gorm:"not null"`
	Lama            string `json:"lama" gorm:"not null"`
	TanggalMulai    string `json:"tanggal_mulai" gorm:"not null"`
	TanggalSelesai  string `json:"tanggal_selesai" gorm:"not null"`
	KepalaPuskesmas string `json:"kepala_puskesmas" gorm:"not null"`
	JumlahPetugas   int    `json:"jumlah_petugas" gorm:"not null"`
	JenisPerjadin   string `json:"jenis_perjadin" gorm:"not null"`
	DalamRangkaId   uint   `json:"dalam_rangka_id" gorm:"not null"`
	AtasNama        string `json:"atas_nama" gorm:"not null"`
	TempatId        uint   `json:"tempat_id" gorm:"not null"`

	Petugas []Pegawai `json:"petugas" gorm:"many2many:surat_pegawai;"`
}

func (Surat) TableName() string {
	return "surat"
}

File: backend/app/Models/Surat.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Surat extends Model
{
    use HasFactory;

    protected $table = 'surat';

    protected $fillable = [
        'nomor_surat',
        'status',
        'tanggal_buat',
        'lama',
        'tanggal_mulai',
        'tanggal_selesai',
        'kepala_puskesmas',
        'jumlah_petugas',
        'jenis_perjadin',
        'dalam_rangka_id',
        'tempat_id',
        'atas_nama',
    ];

    /**
     * Relasi Many-to-Many ke Pegawai
     */
    public function petugas()
    {
        return $this->belongsToMany(Pegawai::class, 'surat_pegawai', 'surat_id', 'pegawai_id');
    }

    /**
     * Relasi One-to-Many ke Lpd
     */
    public function lpd()
    {
        return $this->hasMany(Lpd::class, 'surat_id');
    }

    /**
     * Jika dalam_rangka_id refer ke tabel lain
     */
    public function dalamRangka()
    {
        return $this->belongsTo(DalamRangka::class, 'dalam_rangka_id');
    }

    /**
     * Jika tempat_id refer ke tabel lain
     */
    public function tempat()
    {
        return $this->belongsTo(Tempat::class, 'tempat_id');
    }
}
